extensions/xslt/runtime/src/main/java/org/apache/camel/quarkus/component/xslt/CamelXsltRecorder.java

```diff
@@ -52,6 +55,23 @@ public void addRuntimeUriResolverEntry(RuntimeValue<RuntimeUriResolver.Builder>
         builder.getValue().entry(templateUri, transletClassName);
     }
 
+    static class QuarkusXsltUriResolverFactory extends DefaultXsltUriResolverFactory {
+        private final RuntimeUriResolver uriResolver;
+
+        public QuarkusXsltUriResolverFactory(RuntimeUriResolver uriResolver) {
+            this.uriResolver = uriResolver;
+        }
+
+        @Override
+        public URIResolver createUriResolver(CamelContext camelContext, String resourceUri) {
+            if (uriResolver.getTransletClassName(resourceUri) != null) {
+                return uriResolver;
+            } else {
+                return super.createUriResolver(camelContext, resourceUri);
+            }
```

Comment:
A comment would be nice about what these two options are supposed to do. The first one is supposed to catch the cases where we compile the translet at build time, right? Which is for `classpath:` XSLT resources in both JVM and native mode?
The `else` branch is for all other cases, right?

Reply:
yeah, it is right.